analyzer/src/main/kotlin/managers/utils/DotNetSupport.kt

```diff
@@ -41,9 +47,45 @@ import com.here.ort.utils.textValueOrEmpty
 
 import okhttp3.Request
 
+import java.io.File
 import java.io.IOException
 import java.net.HttpURLConnection
 
+abstract class XmlPackageReferenceMapper {
+    protected val mapper = XmlMapper().registerKotlinModule()
+
+    abstract fun mapPackageReferences(definitionFile: File): Map<String, String>
+}
+
+fun PackageManager.resolveDotNetDependencies(
+    definitionFile: File,
+    mapper: XmlPackageReferenceMapper
+): ProjectAnalyzerResult? {
```

Comment:
I wonder if it would be better to make this function a member function of `DotNetSupport` to have one less layer between the support class and the package manager implementations. It would also match what I do in https://github.com/heremaps/oss-review-toolkit/pull/1593 with `PythonSupport`.

Reply:
The problem is that this function needs access to the ` PackageManager`'s `managerName` and `analysisRoot`...

Reply:
They could be passed as parameters, but I'll approve anyway.